hibernate-ogm/playsport-demo/src/main/java/org/hibernate/ogm/demo/playsport/core/repo/AthleteRepo.java

```diff
@@ -0,0 +1,97 @@
+/*
+ * Copyright 2017 Red Hat, Inc. and/or its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+*/
+package org.hibernate.ogm.demo.playsport.core.repo;
+
+import java.util.List;
+import javax.ejb.Stateless;
+import javax.inject.Inject;
+import javax.persistence.EntityManager;
+import javax.validation.Valid;
+
+import org.apache.lucene.search.Query;
+import org.hibernate.ogm.demo.playsport.core.entity.Athlete;
+import org.hibernate.ogm.demo.playsport.core.entity.Club;
+import org.hibernate.search.jpa.FullTextEntityManager;
+import org.hibernate.search.jpa.Search;
+import org.hibernate.search.query.dsl.QueryBuilder;
+
+/**
+ * @author Fabio Massimo Ercoli (C) 2017 Red Hat Inc.
+ */
+@Stateless
+public class AthleteRepo {
+
+    @Inject
+    private EntityManager em;
+
+    public void add(@Valid Athlete athlete) {
+        em.persist(athlete);
+    }
+
+    public List<Athlete> findAll() {
+        return em.createQuery("select c from Athlete c", Athlete.class).getResultList();
```

Comment:
It seems that it works well if the indexes are consistent with the entries of the caches. Using hibernate.search.default.directory_provider ram I never had any problems.

Reply:
ah right, I forgot that all dialects can use the automatic translation form JPQL to Hibernate Search queries. FYI we still need to implement proper query capabilities for the Hot Rod dialect, so that it doesn't have to maintain local indexes.